### ISSUE TITLE
Nightly 2020-04-08 broke the v0.9.5 builds

### ISSUE THREAD

**Issue by vertexclique** [AltSysrq/proptest#186]
```
latest update on 2020-04-08, rust version 1.44.0-nightly (42abbd887 2020-04-07)
```
It looks like with this nightly ver. builds might have been failing.
I haven't had deep look at that but it seems like overarching lifetimes are the issue.

```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'a` due to conflicting requirements
##[error]  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proptest-0.9.5/src/arbitrary/_core/iter.rs:47:23
   |
47 |         base.prop_map(Iterator::cloned).boxed()
   |                       ^^^^^^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime `'a` as defined on the impl at 36:9...
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proptest-0.9.5/src/arbitrary/_core/iter.rs:36:9
   |
36 |         'a,
   |         ^^
note: ...so that the types are compatible
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proptest-0.9.5/src/arbitrary/_core/iter.rs:47:23
   |
47 |         base.prop_map(Iterator::cloned).boxed()
   |                       ^^^^^^^^^^^^^^^^
   = note: expected  `&'a T`
              found  `&T`
   = note: but, the lifetime must be valid for the static lifetime...
note: ...so that the type `strategy::map::Map<S, fn(A) -> core::iter::Cloned<A> {<A as core::iter::Iterator>::cloned::<'_, T>}>` will meet its required lifetime bounds
  --> /home/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/proptest-0.9.5/src/arbitrary/_core/iter.rs:47:41
   |
47 |         base.prop_map(Iterator::cloned).boxed()
   |                                         ^^^^^

error: aborting due to previous error
```

**Comment by Centril** [comment]
The regression was reported to rust-lang/rust in https://github.com/rust-lang/rust/issues/70917.

**Comment by Centril** [comment]
I've published 0.9.6, and https://github.com/AltSysrq/proptest/pull/187 also merges the fix there into master.

**Comment by vertexclique** [comment]
Thanks
